src/test/record/test_frequency.py

```diff
@@ -0,0 +1,229 @@
+        assert result == expect_raw
+
+    def test_one_column_default_case(self):
+        records_raw = [
+            {'timestamp': 0},
+            {'timestamp': 1},
+            {'timestamp': 11},
+            {'timestamp': 12},
+            {'timestamp': 14},
+        ]
+        columns = [ColumnValue('timestamp')]
+        records = create_records(records_raw, columns)
+
+        frequency = Frequency(records)
+
+        expect_raw = [
+            {'timestamp': 0, 'frequency': 2},
+            {'timestamp': 10, 'frequency': 3}
+        ]
+        result = to_dict(frequency.to_records(interval_ns=10))
+        assert result == expect_raw
+
+    def test_specify_interval_ns_case(self):
+        records_raw = [
+            {'timestamp': 1000000000},
+            {'timestamp': 1000500000},
+            {'timestamp': 2000000000},
+            {'timestamp': 2000300000},
+            {'timestamp': 3000000000},
+        ]
+        columns = [ColumnValue('timestamp')]
+        records = create_records(records_raw, columns)
+
+        frequency = Frequency(records)
+
+        expect_raw = [
+            {'timestamp': 1000000000, 'frequency': 4},
+            {'timestamp': 3000000000, 'frequency': 1}
+        ]
+        result = to_dict(frequency.to_records(interval_ns=2000000000))
+        assert result == expect_raw
+
+    def test_specify_base_timestamp_case(self):
+        records_raw = [
+            {'timestamp': 5},
+            {'timestamp': 6},
+            {'timestamp': 11},
+            {'timestamp': 12},
+            {'timestamp': 15},
+        ]
+        columns = [ColumnValue('timestamp')]
+        records = create_records(records_raw, columns)
+
+        frequency = Frequency(records)
+
+        expect_raw = [
+            {'timestamp': 4, 'frequency': 4},
+            {'timestamp': 14, 'frequency': 1}
+        ]
+        result = to_dict(frequency.to_records(interval_ns=10,
+                                              base_timestamp=4))
+        assert result == expect_raw
+
+    def test_two_column_default_case(self):
+        records_raw = [
+            {'timestamp1': 0, 'timestamp2': 2},
+            {'timestamp1': 3, 'timestamp2': 4},
+            {'timestamp1': 11, 'timestamp2': 12},
+            {'timestamp1': 13, 'timestamp2': 14},
+            {'timestamp1': 15, 'timestamp2': 16},
+        ]
+        columns = [ColumnValue('timestamp1'), ColumnValue('timestamp2')]
+        records = create_records(records_raw, columns)
+
+        frequency = Frequency(records)
+
+        expect_raw = [
+            {'timestamp1': 0, 'frequency': 2},
+            {'timestamp1': 10, 'frequency': 3}
+        ]
+        result = to_dict(frequency.to_records(interval_ns=10))
+        assert result == expect_raw
+
+    def test_specify_target_column_case(self):
+        records_raw = [
+            {'timestamp1': 0, 'timestamp2': 2},
+            {'timestamp1': 3, 'timestamp2': 4},
+            {'timestamp1': 11, 'timestamp2': 12},
```

Comment:
To confirm whether timestamp2 was used, following input seems better.

```suggestion
            {'timestamp1': 5, 'timestamp2': 12},
```

Reply:
I have accepted the suggestion :bow: 
https://github.com/tier4/CARET_analyze/pull/116/commits/9324f170b0030ffb5522d9209cf3f99196f149ef

Reply:
@hsgwa